packages/console/src/pages/EnterpriseSso/SsoConnectorLogo/index.tsx

```diff
@@ -23,9 +23,11 @@ const pickLogoForCurrentTheme = (
   branding: SsoConnectorWithProviderConfig['branding']
 ): string => {
   // Need to use `||` here since `??` operator can not avoid empty strings.
-  // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
-  const configuredLogo = isDarkMode ? branding.darkLogo : branding.logo || branding.darkLogo;
-  const builtInLogo = isDarkMode ? logoDark : logo || logoDark;
+  // Since `logo` and `darkLogo` are both optional, when it is dark mode and `darkLogo` is not configured, should fallback to `logo`.
+  const configuredLogo =
+    // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+    (isDarkMode ? branding.darkLogo : branding.logo) || branding.darkLogo || branding.logo;
+  const builtInLogo = isDarkMode ? logoDark : logo || logoDark || logo;
```

Comment:
This reads a bit odd... So the light `branding.logo` has higher priority than the `logoDark` in dark mode?

Reply:
Good call, should wrap the ternary calculation with brackets.

Reply:
The fallback logic of the display of the logo could not be easy to understand. Let me add unit test cases to deal with it since there should be 2^3=8 different cases.